linera-base/src/identifiers.rs

```diff
@@ -574,6 +584,89 @@ impl std::str::FromStr for Owner {
     }
 }
 
+impl Serialize for AccountOwner {
+    fn serialize<S: serde::ser::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        if serializer.is_human_readable() {
+            serializer.serialize_str(&self.to_string())
+        } else {
+            match *self {
+                AccountOwner::Application(app_id) => {
+                    serializer.serialize_newtype_variant("AccountOwner", 0, "Application", &app_id)
+                }
+                AccountOwner::User(owner) => {
+                    serializer.serialize_newtype_variant("AccountOwner", 1, "User", &owner)
+                }
```

Comment:
You could use `Foo` here, too. Move it out of `deserialize` (but not `pub`, of course), call it e.g. `AccountOwnerSerdeDerive` and derive both `Serialize` and `Deserialize` for it.

That way we don't need string literals that match the variant names.